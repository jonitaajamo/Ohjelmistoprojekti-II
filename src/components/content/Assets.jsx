import React from "react";

const Assets = () => (
  <article className="tile is-child notification">
    <p className="title">Assets</p>
    <p className="subtitle">Assets in selected country</p>
  </article>
);

export default Assets;
