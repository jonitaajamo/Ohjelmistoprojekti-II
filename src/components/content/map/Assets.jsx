import React, { Component } from "react";

export default class Assets extends Component {
  constructor() {
    super();
    this.state = {
      currentMonth: ""
    };
  }

  capitalize(string) {
    return string.charAt(0).toUpperCase() + string.slice(1);
  }

  Loading(what) {
    return (
      <tr>
        <td>Loading {what}...</td>
      </tr>
    );
  }

  renderTopInstruments() {
    return this.props.topInstruments.map((item, key) => {
      return (
        <tr key={key}>
          <td>{key + 1}</td>
          <td>{item.secName}</td>
          <td>{item.totalMarketValue}</td>
          <td>{item.weightInClass * 100 + "%"}</td>
        </tr>
      );
    });
  }

  render() {
    const assetTableData =
      this.props.geographicalWeightData.length === 0
        ? this.Loading("Assets")
        : this.props.geographicalWeightData.map((item, key) => {
            if (key === this.props.month) {
              const innerLoop = item.assetClasses.map((innerItem, innerKey) => {
                const innerInnerLoop = innerItem.weights.map(
                  (innerInnerItem, innerInnerKey) => {
                    if (
                      innerInnerItem.countryId ===
                      Number(this.props.geographyId)
                    ) {
                      return (
                        <tr key={innerKey}>
                          <td>{this.capitalize(innerItem.class)}</td>
                          <td>{innerInnerItem.marketValue + "â‚¬"}</td>
                        </tr>
                      );
                    } else {
                      return null;
                    }
                  }
                );
                return innerInnerLoop;
              });
              return innerLoop;
            } else {
              return null;
            }
          });
    const assetTableStyle = {
      display: !this.props.clickedGeographyName ? "none" : "block"
    };

    return (
      <div className="tile is-vertical">
        <article className="box tile is-child notification has-accent notification">
          <p className="title">Assets</p>
          <p className="subtitle">
            {!this.props.clickedGeographyName
              ? "World"
              : this.props.clickedGeographyName}

            {!this.props.clickedGeographyName
              ? ""
              : " (" + this.props.currency + ")"}

            <br />
            {!this.props.geographicalWeightData.length
              ? this.Loading("Month")
              : this.props.geographicalWeightData[this.props.month].month}
          </p>
          <div className="centered-table" style={assetTableStyle}>
            <table className="table is-fullwidth is-hoverable is-bordered is-striped is-narrow">
              <thead>
                <tr>
                  <th>Asset name</th>
                  <th>Value</th>
                </tr>
              </thead>
              <tbody>{assetTableData}</tbody>
            </table>
          </div>
        </article>
        <article className="box tile is-child notification has-accent notification">
          <p className="title">Top 10 Instruments</p>
          <table className="table is-fullwidth is-hoverable is-bordered is-striped is-narrow">
            <thead>
              <tr>
                <th />
                <th>Instrument name</th>
                <th>Value</th>
                <th>Weight</th>
              </tr>
            </thead>
            <tbody>{this.renderTopInstruments()}</tbody>
          </table>
        </article>
      </div>
    );
  }
}
