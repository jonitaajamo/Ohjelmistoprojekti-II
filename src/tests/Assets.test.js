import React from "react";
import Assets from "../components/content/map/Assets";

describe("Assets", () => {
  const props = {
    geographicalWeightData: [
      {
        month: "2012-01",
        assetClasses: [
          {
            class: "all",
            weights: [
              { countryId: 840, weight: 0.7, marketValue: 14000000 },
              { countryId: 246, weight: 0.2, marketValue: 4000000 },
              { countryId: 826, weight: 0.05, marketValue: 1000000 },
              { countryId: 643, weight: 0.05, marketValue: 1000000 }
            ]
          }
        ]
      }
    ],
    topInstruments: [
      {
        secId: 1236452364,
        secName: "Woahtek",
        weightInClass: 0.4,
        marketValueInClass: 8000000,
        classWeightInPortfolio: 0.08,
        totalMarketValue: 8000000
      }
    ],
    month: 0,
    currency: "EUR"
  };

  it("renders correctly", () => {
    const assets = shallow(<Assets {...props} />);
    expect(assets).toMatchSnapshot();
  });

  it("displays loading text", () => {
    const assets = shallow(
      <Assets geographicalWeightData={[]} topInstruments={[]} />
    );
    expect(
      assets
        .find("tr")
        .at(0)
        .text()
    ).toEqual("Loading Month...");
  });

  it("displays month", () => {
    const assets = shallow(<Assets {...props} />);
    const subtitle = assets.find("p.subtitle");
    expect(subtitle.text()).toContain("2012-01");
  });

  it("renders top instruments by calling method", () => {
    const spy = jest.spyOn(Assets.prototype, "renderTopInstruments");
    const assets = shallow(<Assets {...props} />);
    expect(spy).toHaveBeenCalledTimes(1);
    expect(
      assets
        .find("td")
        .at(1)
        .text()
    ).toEqual("Woahtek");
  });

  it("displays selected country name and currency", () => {
    const assets = shallow(
      <Assets {...props} clickedGeographyName="Finland" />
    );
    const subtitle = assets.find("p.subtitle");
    expect(subtitle.text()).toContain("Finland (EUR)");
  });
});
