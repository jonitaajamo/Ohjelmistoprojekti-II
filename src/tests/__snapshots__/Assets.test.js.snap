// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assets renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Assets
    currency="EUR"
    geographicalWeightData={
        Array [
            Object {
              "assetClasses": Array [
                Object {
                  "class": "all",
                  "weights": Array [
                    Object {
                      "countryId": 840,
                      "marketValue": 14000000,
                      "weight": 0.7,
                    },
                    Object {
                      "countryId": 246,
                      "marketValue": 4000000,
                      "weight": 0.2,
                    },
                    Object {
                      "countryId": 826,
                      "marketValue": 1000000,
                      "weight": 0.05,
                    },
                    Object {
                      "countryId": 643,
                      "marketValue": 1000000,
                      "weight": 0.05,
                    },
                  ],
                },
              ],
              "month": "2012-01",
            },
          ]
    }
    month={0}
    selectedAsset={0}
    topInstruments={
        Array [
            Object {
              "classWeightInPortfolio": 0.08,
              "marketValueInClass": 8000000,
              "secId": 1236452364,
              "secName": "Woahtek",
              "totalMarketValue": 8000000,
              "weightInClass": 0.4,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article
          className="box tile is-child notification has-accent notification"
>
          <p
                    className="title"
          >
                    Assets
          </p>
          <p
                    className="subtitle"
          >
                    World
                    
                    <br />
                    2012-01
          </p>
          <div
                    className="centered-table"
                    style={
                              Object {
                                        "display": "none",
                                      }
                    }
          >
                    <table
                              className="table is-fullwidth is-hoverable is-bordered is-striped is-narrow"
                    >
                              <thead />
                              <tbody />
                    </table>
          </div>
</article>,
        <article
          className="box tile is-child notification has-accent notification"
>
          <p
                    className="title"
          >
                    Top 10 Instruments
          </p>
          <table
                    className="table is-fullwidth is-hoverable is-bordered is-striped is-narrow"
          >
                    <thead>
                              <tr>
                                        <th />
                                        <th>
                                                  Instrument name
                                        </th>
                                        <th>
                                                  Value
                                        </th>
                                        <th>
                                                  Weight
                                        </th>
                              </tr>
                    </thead>
                    <tbody>
                              <tr>
                                        <td>
                                                  1
                                        </td>
                                        <td>
                                                  Woahtek
                                        </td>
                                        <td>
                                                  8000000
                                                  â‚¬
                                        </td>
                                        <td>
                                                  40%
                                        </td>
                              </tr>
                    </tbody>
          </table>
</article>,
      ],
      "className": "tile is-vertical",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="title"
>
              Assets
</p>,
            <p
              className="subtitle"
>
              World
              
              <br />
              2012-01
</p>,
            <div
              className="centered-table"
              style={
                            Object {
                                          "display": "none",
                                        }
              }
>
              <table
                            className="table is-fullwidth is-hoverable is-bordered is-striped is-narrow"
              >
                            <thead />
                            <tbody />
              </table>
</div>,
          ],
          "className": "box tile is-child notification has-accent notification",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Assets",
              "className": "title",
            },
            "ref": null,
            "rendered": "Assets",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "World",
                "",
                <br />,
                "2012-01",
              ],
              "className": "subtitle",
            },
            "ref": null,
            "rendered": Array [
              "World",
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "2012-01",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <table
                className="table is-fullwidth is-hoverable is-bordered is-striped is-narrow"
>
                <thead />
                <tbody />
</table>,
              "className": "centered-table",
              "style": Object {
                "display": "none",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <thead />,
                  <tbody />,
                ],
                "className": "table is-fullwidth is-hoverable is-bordered is-striped is-narrow",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "thead",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "tbody",
                },
              ],
              "type": "table",
            },
            "type": "div",
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="title"
>
              Top 10 Instruments
</p>,
            <table
              className="table is-fullwidth is-hoverable is-bordered is-striped is-narrow"
>
              <thead>
                            <tr>
                                          <th />
                                          <th>
                                                        Instrument name
                                          </th>
                                          <th>
                                                        Value
                                          </th>
                                          <th>
                                                        Weight
                                          </th>
                            </tr>
              </thead>
              <tbody>
                            <tr>
                                          <td>
                                                        1
                                          </td>
                                          <td>
                                                        Woahtek
                                          </td>
                                          <td>
                                                        8000000
                                                        â‚¬
                                          </td>
                                          <td>
                                                        40%
                                          </td>
                            </tr>
              </tbody>
</table>,
          ],
          "className": "box tile is-child notification has-accent notification",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Top 10 Instruments",
              "className": "title",
            },
            "ref": null,
            "rendered": "Top 10 Instruments",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <thead>
                  <tr>
                                    <th />
                                    <th>
                                                      Instrument name
                                    </th>
                                    <th>
                                                      Value
                                    </th>
                                    <th>
                                                      Weight
                                    </th>
                  </tr>
</thead>,
                <tbody>
                  <tr>
                                    <td>
                                                      1
                                    </td>
                                    <td>
                                                      Woahtek
                                    </td>
                                    <td>
                                                      8000000
                                                      â‚¬
                                    </td>
                                    <td>
                                                      40%
                                    </td>
                  </tr>
</tbody>,
              ],
              "className": "table is-fullwidth is-hoverable is-bordered is-striped is-narrow",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <tr>
                    <th />
                    <th>
                                        Instrument name
                    </th>
                    <th>
                                        Value
                    </th>
                    <th>
                                        Weight
                    </th>
</tr>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <th />,
                      <th>
                        Instrument name
</th>,
                      <th>
                        Value
</th>,
                      <th>
                        Weight
</th>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Instrument name",
                      },
                      "ref": null,
                      "rendered": "Instrument name",
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Value",
                      },
                      "ref": null,
                      "rendered": "Value",
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Weight",
                      },
                      "ref": null,
                      "rendered": "Weight",
                      "type": "th",
                    },
                  ],
                  "type": "tr",
                },
                "type": "thead",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <tr>
                      <td>
                                            1
                      </td>
                      <td>
                                            Woahtek
                      </td>
                      <td>
                                            8000000
                                            â‚¬
                      </td>
                      <td>
                                            40%
                      </td>
</tr>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "0",
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <td>
                          1
</td>,
                        <td>
                          Woahtek
</td>,
                        <td>
                          8000000
                          â‚¬
</td>,
                        <td>
                          40%
</td>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": 1,
                        },
                        "ref": null,
                        "rendered": 1,
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Woahtek",
                        },
                        "ref": null,
                        "rendered": "Woahtek",
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            8000000,
                            "â‚¬",
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          8000000,
                          "â‚¬",
                        ],
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "40%",
                        },
                        "ref": null,
                        "rendered": "40%",
                        "type": "td",
                      },
                    ],
                    "type": "tr",
                  },
                ],
                "type": "tbody",
              },
            ],
            "type": "table",
          },
        ],
        "type": "article",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article
            className="box tile is-child notification has-accent notification"
>
            <p
                        className="title"
            >
                        Assets
            </p>
            <p
                        className="subtitle"
            >
                        World
                        
                        <br />
                        2012-01
            </p>
            <div
                        className="centered-table"
                        style={
                                    Object {
                                                "display": "none",
                                              }
                        }
            >
                        <table
                                    className="table is-fullwidth is-hoverable is-bordered is-striped is-narrow"
                        >
                                    <thead />
                                    <tbody />
                        </table>
            </div>
</article>,
          <article
            className="box tile is-child notification has-accent notification"
>
            <p
                        className="title"
            >
                        Top 10 Instruments
            </p>
            <table
                        className="table is-fullwidth is-hoverable is-bordered is-striped is-narrow"
            >
                        <thead>
                                    <tr>
                                                <th />
                                                <th>
                                                            Instrument name
                                                </th>
                                                <th>
                                                            Value
                                                </th>
                                                <th>
                                                            Weight
                                                </th>
                                    </tr>
                        </thead>
                        <tbody>
                                    <tr>
                                                <td>
                                                            1
                                                </td>
                                                <td>
                                                            Woahtek
                                                </td>
                                                <td>
                                                            8000000
                                                            â‚¬
                                                </td>
                                                <td>
                                                            40%
                                                </td>
                                    </tr>
                        </tbody>
            </table>
</article>,
        ],
        "className": "tile is-vertical",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="title"
>
                Assets
</p>,
              <p
                className="subtitle"
>
                World
                
                <br />
                2012-01
</p>,
              <div
                className="centered-table"
                style={
                                Object {
                                                "display": "none",
                                              }
                }
>
                <table
                                className="table is-fullwidth is-hoverable is-bordered is-striped is-narrow"
                >
                                <thead />
                                <tbody />
                </table>
</div>,
            ],
            "className": "box tile is-child notification has-accent notification",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Assets",
                "className": "title",
              },
              "ref": null,
              "rendered": "Assets",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "World",
                  "",
                  <br />,
                  "2012-01",
                ],
                "className": "subtitle",
              },
              "ref": null,
              "rendered": Array [
                "World",
                "",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "2012-01",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <table
                  className="table is-fullwidth is-hoverable is-bordered is-striped is-narrow"
>
                  <thead />
                  <tbody />
</table>,
                "className": "centered-table",
                "style": Object {
                  "display": "none",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <thead />,
                    <tbody />,
                  ],
                  "className": "table is-fullwidth is-hoverable is-bordered is-striped is-narrow",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": null,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "thead",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": null,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "tbody",
                  },
                ],
                "type": "table",
              },
              "type": "div",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="title"
>
                Top 10 Instruments
</p>,
              <table
                className="table is-fullwidth is-hoverable is-bordered is-striped is-narrow"
>
                <thead>
                                <tr>
                                                <th />
                                                <th>
                                                                Instrument name
                                                </th>
                                                <th>
                                                                Value
                                                </th>
                                                <th>
                                                                Weight
                                                </th>
                                </tr>
                </thead>
                <tbody>
                                <tr>
                                                <td>
                                                                1
                                                </td>
                                                <td>
                                                                Woahtek
                                                </td>
                                                <td>
                                                                8000000
                                                                â‚¬
                                                </td>
                                                <td>
                                                                40%
                                                </td>
                                </tr>
                </tbody>
</table>,
            ],
            "className": "box tile is-child notification has-accent notification",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Top 10 Instruments",
                "className": "title",
              },
              "ref": null,
              "rendered": "Top 10 Instruments",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <thead>
                    <tr>
                                        <th />
                                        <th>
                                                            Instrument name
                                        </th>
                                        <th>
                                                            Value
                                        </th>
                                        <th>
                                                            Weight
                                        </th>
                    </tr>
</thead>,
                  <tbody>
                    <tr>
                                        <td>
                                                            1
                                        </td>
                                        <td>
                                                            Woahtek
                                        </td>
                                        <td>
                                                            8000000
                                                            â‚¬
                                        </td>
                                        <td>
                                                            40%
                                        </td>
                    </tr>
</tbody>,
                ],
                "className": "table is-fullwidth is-hoverable is-bordered is-striped is-narrow",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <tr>
                      <th />
                      <th>
                                            Instrument name
                      </th>
                      <th>
                                            Value
                      </th>
                      <th>
                                            Weight
                      </th>
</tr>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <th />,
                        <th>
                          Instrument name
</th>,
                        <th>
                          Value
</th>,
                        <th>
                          Weight
</th>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": "th",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Instrument name",
                        },
                        "ref": null,
                        "rendered": "Instrument name",
                        "type": "th",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Value",
                        },
                        "ref": null,
                        "rendered": "Value",
                        "type": "th",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Weight",
                        },
                        "ref": null,
                        "rendered": "Weight",
                        "type": "th",
                      },
                    ],
                    "type": "tr",
                  },
                  "type": "thead",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <tr>
                        <td>
                                                1
                        </td>
                        <td>
                                                Woahtek
                        </td>
                        <td>
                                                8000000
                                                â‚¬
                        </td>
                        <td>
                                                40%
                        </td>
</tr>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": "0",
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <td>
                            1
</td>,
                          <td>
                            Woahtek
</td>,
                          <td>
                            8000000
                            â‚¬
</td>,
                          <td>
                            40%
</td>,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": 1,
                          },
                          "ref": null,
                          "rendered": 1,
                          "type": "td",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Woahtek",
                          },
                          "ref": null,
                          "rendered": "Woahtek",
                          "type": "td",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": Array [
                              8000000,
                              "â‚¬",
                            ],
                          },
                          "ref": null,
                          "rendered": Array [
                            8000000,
                            "â‚¬",
                          ],
                          "type": "td",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "40%",
                          },
                          "ref": null,
                          "rendered": "40%",
                          "type": "td",
                        },
                      ],
                      "type": "tr",
                    },
                  ],
                  "type": "tbody",
                },
              ],
              "type": "table",
            },
          ],
          "type": "article",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
